stock_storage_type/models/stock_location.py

```diff
@@ -361,10 +365,37 @@ def get_storage_locations(self, products=None):
         locations = self.browse()
         if self.pack_putaway_strategy == "none":
             locations = self
-        elif self.pack_putaway_strategy == "ordered_locations":
-            locations = self._get_ordered_leaf_locations()
+        else:
+            locations = self._get_sorted_leaf_locations(products)
         return locations
 
+    def _get_sorted_leaf_locations_orderby(self, products):
+        orderby = []
+        if self.pack_putaway_strategy == "ordered_locations":
+            orderby = [
+                "CASE WHEN max_height > 0 THEN max_height ELSE 'Infinity' END",
```

Comment:
We could create a computed field to replace the `CASE THEN ELSE` syntax.

Reply:
It is already a stored computed field but the value is 0 when not defined and should be put after positive values

Reply:
If we encounter perf issue into the sorting by postgres, we could have a `max_height_sort_indice` computed from max_height and create an index `(max_height_sort_indice, pack_putaway_sequence, name)`

Reply:
@lmignon You could also create your index with
CREATE INDEX .... ON stock_location((CASE WHEN max_height > 0 THEN max_height ELSE 'Infinity' END));

Reply:
@jbaudoux I'd add a TODO comment

Reply:
@simahawk A TODO for what? Even with an INDEX and large location table, in my case postgresql favors sql scan probably due to subsequent sorting